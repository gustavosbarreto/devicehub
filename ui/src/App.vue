<template>
<v-app>
    <v-navigation-drawer v-if="isLoggedIn" :clipped="false" fixed v-model="drawer" :mini-variant="false" enable-resize-watcher app>
        <v-container>
            <div class="text-center">
                <v-icon>mdi-console</v-icon>
                <h2 style="font-family: monospace">ShellHub</h2>
                <span class="overline">beta</span>
            </div>
        </v-container>
        <v-list>
            <v-list-item v-for="item in items" :key="item.title" :to="item.path" two-line>
                <v-list-item-action>
                    <v-icon v-text="item.icon"></v-icon>
                </v-list-item-action>

                <v-list-item-content>
                    <v-list-item-title v-text="item.title"></v-list-item-title>
                </v-list-item-content>
            </v-list-item>
        </v-list>
    </v-navigation-drawer>
    <v-app-bar app elevate-on-scroll color="primary" class="pl-3 pr-4" v-if="isLoggedIn">
        <v-menu transition="scale-transition" origin="top left">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-server</v-icon>
                    My Device Fleet
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>
        </v-menu>
        <v-spacer></v-spacer>
        <v-chip>
            <v-icon>help</v-icon>
        </v-chip>
        <v-chip>
            <v-icon>notifications</v-icon>
        </v-chip>
        <v-menu transition="scale-transition" origin="top right">
            <template v-slot:activator="{ on }">
                <v-chip v-on="on">
                    <v-icon left>mdi-account</v-icon>
                    {{ $store.getters["auth/currentUser"] }}
                    <v-icon right>mdi-chevron-down</v-icon>
                </v-chip>
            </template>

            <v-card>
                <v-list-item three-line>
                    <v-list-item-content>
                        <v-list-item-title class="mb-1">Tenant ID</v-list-item-title>
                        <v-list-item-subtitle>
                            <v-chip>
                                <span>
                                    {{ tenant }}
                                </span>
                                <v-icon right v-clipboard="tenant" v-clipboard:success="() => { copySnack = true; }">mdi-content-copy</v-icon>
                            </v-chip>
                        </v-list-item-subtitle>
                    </v-list-item-content>
                </v-list-item>

                <v-card-actions>
                    <v-btn small text @click="logout()">
                        Logout
                    </v-btn>
                </v-card-actions>
            </v-card>
        </v-menu>
    </v-app-bar>
    <v-content>
        <v-container class="pa-8" fluid>
            <router-view></router-view>
        </v-container>
        <v-snackbar v-model="copySnack" :timeout=3000>Tenant ID copied to clipboard</v-snackbar>
    </v-content>
</v-app>
</template>

<script>
export default {
  name: "App",

  methods: {
    logout() {
      this.$store.dispatch("auth/logout").then(() => {
        this.$router.push("/login");
      });
    }
  },

  computed: {
    tenant() {
      return this.$store.getters["auth/tenant"];
    },

    isLoggedIn() {
      return this.$store.getters["auth/isLoggedIn"];
    }
  },

  data() {
    return {
      drawer: true,
      clipped: false,
      copySnack: false,
      items: [
        {
          icon: "dashboard",
          title: "Dashboard",
          path: "/"
        },
        {
          icon: "devices",
          title: "Devices",
          path: "/devices"
        },
        {
          icon: "history",
          title: "Sessions",
          path: "/sessions"
        }
      ]
    };
  }
};
</script>

<style>
.v-list-active {
  border-left: 4px solid var(--v-primary-base);
}

.text-shadow {
  text-shadow: #000 0 0 6px;
  color: transparent;
}
</style>
