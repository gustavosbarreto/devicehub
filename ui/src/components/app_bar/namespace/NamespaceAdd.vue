<template>
  <fragment>
    <v-list-item-title>
      <v-dialog
        v-model="showAddNamespace"
        max-width="450"
        @click:outside="cancel"
      >
        <v-card>
          <v-card-title class="headline grey lighten-2 text-center">
            Enter Namespace
          </v-card-title>
          <ValidationObserver
            ref="obs"
            v-slot="{ passes }"
          >
            <v-card-text class="caption mb-0">
              <ValidationProvider
                v-slot="{ errors }"
                name="namespace"
                rules="required|rfc1123"
              >
                <v-text-field
                  v-model="namespaceName"
                  label="Namespace"
                  :error-messages="errors"
                  require
                />
              </ValidationProvider>
            </v-card-text>
            <v-card-actions>
              <v-spacer />
              <v-btn
                text
                @click="cancel"
              >
                Close
              </v-btn>
              <v-btn
                color="primary"
                text
                @click="passes(addNamespace)"
              >
                Add
              </v-btn>
            </v-card-actions>
          </ValidationObserver>
        </v-card>
      </v-dialog>
    </v-list-item-title>
  </fragment>
</template>

<script>

import {
  ValidationObserver,
  ValidationProvider,
} from 'vee-validate';

export default {
  name: 'NamespaceAdd',

  components: {
    ValidationProvider,
    ValidationObserver,
  },

  props: {
    firstNamespace: {
      type: Boolean,
      default: false,
    },

    show: {
      type: Boolean,
      required: true,
    },
  },

  data() {
    return {
      dialog: false,
      namespaceName: '',
    };
  },

  computed: {
    showAddNamespace: {
      get() {
        return this.show;
      },

      set(value) {
        this.$emit('show', value);
      },
    },
  },

  methods: {
    cancel() {
      this.dialog = false;
      this.$refs.obs.reset();
      this.namespaceName = '';
      this.$emit('update:show', false);
    },

    async switchIn(tenant) {
      try {
        await this.$store.dispatch('namespaces/switchNamespace', {
          tenant_id: tenant,
        });
        window.location.reload();
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorLoading', this.$errors.namespaceSwitch);
      }
    },

    async addNamespace() {
      try {
        const response = await this.$store.dispatch('namespaces/post', {
          name: this.namespaceName,
        });
        if (this.$props.firstNamespace) {
          await this.switchIn(response.data.tenant_id);
        } else {
          await this.$store.dispatch('namespaces/fetch');
          this.$emit('update:show', false);
        }
        this.dialog = false;
        this.namespaceName = '';
        this.$refs.obs.reset();
        this.$store.dispatch('snackbar/showSnackbarSuccessAction', this.$success.namespaceCreating);
      } catch {
        this.$store.dispatch('snackbar/showSnackbarErrorAction', this.$errors.namespaceCreating);
      }
    },
  },
};

</script>
