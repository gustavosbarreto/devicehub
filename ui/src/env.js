export default {
  /* eslint-disable no-param-reassign */
  install(Vue) {
    const env = {
      isHosted: window.env.HOSTED === 'true',
    };

    Vue.env = env;
    Vue.prototype.$env = env;
  },
  /* eslint-enable no-param-reassign */
};
