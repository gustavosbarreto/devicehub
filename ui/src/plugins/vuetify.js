import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import '@mdi/font/css/materialdesignicons.css';
import 'font-logos/assets/font-logos.css';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
import { fas } from '@fortawesome/free-solid-svg-icons';
import Clipboard from 'v-clipboard';
import Fragment from 'vue-fragment';

//import 'vuetify/dist/vuetify.min.css'
//import '../styles/variables.scss'

Vue.component('font-awesome-icon', FontAwesomeIcon); // Register component globally
library.add(fas); // Include needed icons.

Vue.use(Vuetify);
Vue.use(Clipboard);
Vue.use(Fragment.Plugin);

const DEVICE_ICONS = {
  alpine: 'fl-alpine',
  arch: 'fl-archlinux',
  centos: 'fl-centos',
  coreos: 'fl-coreos',
  debian: 'fl-debian',
  devuan: 'fl-devuan',
  elementary: 'fl-elementary',
  fedora: 'fl-fedora',
  freebsd: 'fl-freebsd',
  gentoo: 'fl-gentoo',
  linuxmint: 'fl-linuxmint',
  mageia: 'fl-mageia',
  manjaro: 'fl-manjaro',
  mandriva: 'fl-mandriva',
  nixos: 'fl-nixos',
  opensuse: 'fl-opensuse',
  rhel: 'fl-redhat',
  sabayon: 'fl-sabayon',
  slackware: 'fl-slackware',
  ubuntu: 'fl-ubuntu',
  raspbian: 'fl-raspberry-pi',
  'ubuntu-core': 'fl-ubuntu',
  void: 'fl-void',
};

export default new Vuetify({
  iconfont: 'md',
  icons:{
    values:DEVICE_ICONS
  },
  theme: {
    dark: false,
    themes: {
      light: {
        primary: '#364576',
        secondary: '#b0bec5',
        accent: '#8c9eff',
        error: '#b71c1c',
      },
    }
  }
});
