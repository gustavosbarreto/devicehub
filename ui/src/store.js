import Vue from 'vue';
import Vuex from 'vuex';
import stats from '@/modules/stats';
import sessions from '@/modules/sessions';
import auth from '@/modules/auth';
import devices from '@/modules/devices';
import modals from '@/modules/modals';
import firewallrules from '@/modules/firewall_rules';
import notifications from '@/modules/notifications';

Vue.use(Vuex);

export default new Vuex.Store({
  modules: {
    devices, modals, stats, sessions, auth, firewallrules, notifications,
  },
});
