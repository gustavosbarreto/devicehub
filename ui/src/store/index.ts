import { InjectionKey } from "vue";
import { createStore, Store, useStore as vuexUseStore } from "vuex";

import { auth, AuthState } from "./modules/auth";
import { apiKeys, ApiKeysState } from "./modules/api_keys";
import { layout, LayoutState } from "./modules/layout";
import { users, UsersState } from "./modules/users";
import { tags, TagsState } from "./modules/tags";
import { stats, StatsState } from "./modules/stats";
import { spinner, SpinnerState } from "./modules/spinner";
import { snackbar, SnackbarState } from "./modules/snackbar";
import { sessions, SessionsState } from "./modules/sessions";
import { security, SecurityState } from "./modules/security";
import { publicKeys, PublicKeysState } from "./modules/public_keys";
import { privateKey, PrivateKeyState } from "./modules/private_key";
import { notifications, NotificationsState } from "./modules/notifications";
import { modal, ModalState } from "./modules/modal";
import { firewallRules, FirewallRulesState } from "./modules/firewall_rules";
import { devices, DevicesState } from "./modules/devices";
import { box, BoxState } from "./modules/box";
import { namespaces, NamespacesState } from "./modules/namespaces";
import { billing } from "./modules/billing";
import { customer, CustomerState } from "./modules/customer";
import { announcement, AnnouncementState } from "./modules/announcement";
import apiPlugin from "./plugins/api";

export interface State {
  auth: AuthState;
  apiKeys: ApiKeysState;
  billing: NamespacesState;
  customer: CustomerState;
  box: BoxState;
  devices: DevicesState;
  firewallRules: FirewallRulesState;
  layout: LayoutState;
  modal: ModalState;
  namespaces: NamespacesState;
  notifications: NotificationsState;
  privateKey: PrivateKeyState;
  publicKeys: PublicKeysState;
  security: SecurityState;
  sessions: SessionsState;
  snackbar: SnackbarState;
  spinner: SpinnerState;
  stats: StatsState;
  tags: TagsState;
  users: UsersState;
  announcement: AnnouncementState;
}

export const key: InjectionKey<Store<State>> = Symbol("store");

export const store = createStore<State>({
  modules: {
    auth,
    apiKeys,
    customer,
    billing,
    box,
    devices,
    firewallRules,
    layout,
    modal,
    namespaces,
    notifications,
    privateKey,
    publicKeys,
    security,
    sessions,
    snackbar,
    spinner,
    stats,
    tags,
    users,
    announcement,
  },
  plugins: [
    apiPlugin,
  ],
});

export function useStore(): Store<State> {
  return vuexUseStore(key);
}
