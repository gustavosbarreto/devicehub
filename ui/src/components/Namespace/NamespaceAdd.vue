<template>
  <div class="ml-2 mr-2">
    <v-btn
      v-if="!firstNamespace"
      block
      :size="isSmall ? 'small' : 'default'"
      color="primary"
      @click="showDialog = true"
    >
      Add Namespace
    </v-btn>

    <v-list-item-title>
      <v-dialog
        v-model="showDialog"
        @click:outside="update"
        :max-width="!openVersion ? '400' : '650'"
        v-bind="$attrs"
      >
        <v-card data-test="namespaceAdd-card" class="bg-v-theme-surface">
          <div v-if="!openVersion">
            <v-card-title class="text-headline bg-primary">
              Enter Namespace
            </v-card-title>

            <v-card-text>
              <v-text-field
                v-model="namespaceName"
                label="Username"
                :error-messages="namespaceNameError"
                required
                variant="underlined"
                data-test="username-text"
              />
            </v-card-text>

            <v-card-actions>
              <v-spacer />
              <v-btn data-test="close-btn" @click="update"> Close </v-btn>

              <v-btn color="primary" data-test="add-btn" @click="addNamespace">
                Add
              </v-btn>
            </v-card-actions>
          </div>
          <div v-else>
            <v-card-title class="bg-primary">
              Add a namespace using the CLI
            </v-card-title>

            <v-card-text class="mt-4 mb-0 pb-1 mb-4">
              <p class="text-body-2">
                In the Community Edition of ShellHub, namespaces must be added using the administration CLI.
                For detailed instructions on how to add namespaces, please refer to the documentation at the ShellHub Administration Guide.
              </p>
              <div id="cli-instructions" class="mt-3 text-body-2">
                <p class="text-caption mb-0 mt-3" data-test="openContentSecond-text">
                  Check the
                  <a
                    :href="'https://docs.shellhub.io/self-hosted/administration'"
                    target="_blank"
                    rel="noopener noreferrer"
                  >ShellHub Administration Guide</a
                  >
                  for more information.
                </p>
              </div>
            </v-card-text>
          </div>
        </v-card>
      </v-dialog>
    </v-list-item-title>
  </div>
</template>

<script setup lang="ts">
import { computed, ref } from "vue";
import * as yup from "yup";
import { useField } from "vee-validate";
import axios, { AxiosError } from "axios";
import {
  INotificationsError,
  INotificationsSuccess,
} from "../../interfaces/INotifications";
import { useStore } from "../../store";
import handleError from "../../utils/handleError";
import { envVariables } from "@/envVariables";

const props = defineProps({
  firstNamespace: {
    type: Boolean,
    default: false,
  },
  show: {
    type: Boolean,
    default: false,
  },
  isSmall: {
    type: Boolean,
    default: false,
  },
  enableSwitchIn: {
    type: Boolean,
    default: false,
  },
});
const emit = defineEmits(["update"]);
const store = useStore();

const showDialog = ref(false);

const {
  value: namespaceName,
  errorMessage: namespaceNameError,
  setErrors: setNamespaceNameError,
  resetField: resetNamespaceName,
} = useField<string>(
  "namespaceName",
  yup
    .string()
    .required()
    .min(3)
    .max(30)
    .matches(/^[^.]*$/, "The name must not contain dots"),
  {
    initialValue: "",
  },
);

const switchIn = async (tenant: string) => {
  try {
    await store.dispatch("namespaces/switchNamespace", {
      tenant_id: tenant,
    });
    window.location.reload();
  } catch (error: unknown) {
    store.dispatch(
      "snackbar/showSnackbarErrorLoading",
      INotificationsError.namespaceSwitch,
    );
    handleError(error);
  }
};

const close = () => {
  showDialog.value = false;
  resetNamespaceName();
};

const update = () => {
  emit("update");
  close();
};

const addNamespace = async () => {
  if (!namespaceNameError.value) {
    try {
      // const tenant = localStorage.getItem("tenant");

      const response = await store.dispatch("namespaces/post", namespaceName.value);

      if (props.firstNamespace || props.enableSwitchIn) {
        await switchIn(response.data.tenant_id);
        close();
      } else {
        await store.dispatch("namespaces/fetch", {
          page: 1,
          perPage: 30,
        });
        update();
      }

      store.dispatch(
        "snackbar/showSnackbarSuccessAction",
        INotificationsSuccess.namespaceCreating,
      );
    } catch (error: unknown) {
      if (axios.isAxiosError(error)) {
        const axiosError = error as AxiosError;
        if (axiosError.response?.status === 400) {
          setNamespaceNameError(
            "Your namespace should be 3-30 characters long",
          );
        } else if (axiosError.response?.status === 403) {
          setNamespaceNameError("Update your plan to create more namespaces");
        } else if (axiosError.response?.status === 409) {
          setNamespaceNameError("namespace already exists");
        }
      } else {
        store.dispatch(
          "snackbar/showSnackbarErrorAction",
          INotificationsError.namespaceCreating,
        );
        handleError(error);
      }
    }
  }
};

const openVersion = computed(() => !envVariables.isCloud || !envVariables.isEnterprise);
</script>
