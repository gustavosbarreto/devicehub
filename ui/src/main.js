import Vue from 'vue';

import Snackbar from '@/components/snackbar/Snackbar';

import App from './App';
import router from './router';
import store from './store';
import env from './env';
import './vee-validate';
import vuetify from './plugins/vuetify';

Vue.config.productionTip = false;

Vue.component('Snackbar', Snackbar);

Vue.use(require('vue-moment'));

Vue.use(env);

new Vue({
  vuetify,
  router,
  store,
  render: (h) => h(App),
}).$mount('#app');
