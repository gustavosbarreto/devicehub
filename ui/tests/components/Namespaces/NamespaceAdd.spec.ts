import { createVuetify } from "vuetify";
import { flushPromises, mount, VueWrapper } from "@vue/test-utils";
import { beforeEach, describe, expect, it, vi } from "vitest";
import NamespaceAdd from "../../../src/components/Namespace/NamespaceAdd.vue";
import { createStore } from "vuex";
import { key } from "../../../src/store";
import routes from "../../../src/router";
import { envVariables } from "../../../src/envVariables";

const show = true;
const firstNamespace = true;

const invalidNamespaces = [
  "'",
  '"',
  "!",
  "@",
  "#",
  "$",
  "%",
  "¨",
  "&",
  "*",
  "(",
  ")",
  "-",
  "_",
  "=",
  "+",
  "´",
  "`",
  "[",
  "{",
  "~",
  "^",
  "]",
  ",",
  "<",
  "..",
  ">",
  ";",
  ":",
  "/",
  "?",
];

const invalidMinAndMaxCharacters = [
  "s",
  "sh",
  "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
  "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
];

const store = createStore({
  state: {},
  getters: {},
  actions: {
    "namespaces/switchNamespace": vi.fn(),
    "namespaces/post": vi.fn(),
    "namespaces/fetch": vi.fn(),
    "snackbar/showSnackbarSuccessAction": vi.fn(),
    "snackbar/showSnackbarErrorAction": vi.fn(),
    "snackbar/showSnackbarErrorLoading": vi.fn(),
  },
});

describe("NamespaceAdd", () => {
  let wrapper: VueWrapper<any>;
  const vuetify = createVuetify();

  ///////
  // In this case, the rendering of the dialog is checked. In which
  // case with the input data it cannot take place.
  ///////

  describe("Doesn't render the dialog", () => {
    beforeEach(() => {
      wrapper = mount(NamespaceAdd, {
        global: {
          plugins: [[store, key], routes, vuetify],
        },
        props: { show: !show, firstNamespace },
        // shallow: true,
      });

      envVariables.isEnterprise = true;
    });

    ///////
    // Component Rendering
    //////

    it("Is a Vue instance", () => {
      expect(wrapper).toBeTruthy();
    });
    it("Renders the component", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data checking
    //////
    it("Data is defined", () => {
      expect(wrapper.vm.$data).toBeDefined();
    });

    it("Receive data in props", () => {
      expect(wrapper.vm.showDialog).toEqual(!show);
      expect(wrapper.vm.firstNamespace).toEqual(firstNamespace);
    });
    it("Compare data with default value", () => {
      expect(wrapper.vm.showDialog).toEqual(false);
      expect(wrapper.vm.namespaceName).toEqual("");
    });
    it("Process data in the computed", () => {
      expect(wrapper.vm.showDialog).toEqual(false);
    });
    //////
    // HTML validation
    //////

    it("Renders the template with data", () => {
      expect(wrapper.find('[data-test="namespaceAdd-card"]').exists()).toBe(
        false
      );
      expect(wrapper.find('[data-test="namespace-text"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="close-btn"]').exists()).toBe(false);
      expect(wrapper.find('[data-test="add-btn"]').exists()).toBe(false);
    });
  });

  ///////
  // In this case, the rendering of the dialog is checked. In which
  // case with the input data it cannot take place.
  ///////

  describe("Dialog", () => {
    beforeEach(() => {
      wrapper = mount(NamespaceAdd, {
        global: {
          plugins: [[store, key], routes, vuetify],
        },
        props: { show: show, firstNamespace },
        shallow: true,
      });
      wrapper.vm.showDialog = true;

      envVariables.isEnterprise = true;
    });

    ///////
    // Component Rendering
    //////

    it("Is a Vue instance", () => {
      expect(wrapper).toBeTruthy();
    });
    it("Renders the component", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    ///////
    // Data checking
    //////
    it("Data is defined", () => {
      expect(wrapper.vm.$data).toBeDefined();
    });

    it("Receive data in props", () => {
      expect(wrapper.vm.showDialog).toEqual(show);
      expect(wrapper.vm.firstNamespace).toEqual(firstNamespace);
    });
    it("Compare data with default value", () => {
      expect(wrapper.vm.namespaceName).toEqual("");
    });
    it("Process data in the computed", () => {
      expect(wrapper.vm.showDialog).toEqual(true);
    });

    //////
    // HTML validation
    //////

    //////
    // In this case, the empty fields are validated.
    //////

    it("Show validation messages", async () => {
      wrapper.vm.namespaceName = "ShelHub";
      wrapper.vm.namespaceName = undefined;
      await flushPromises();
      expect(wrapper.vm.namespaceNameError).toBe("this is a required field");
    });

    it("Show validation messages", async () => {
      wrapper.vm.namespaceName = "ShelHub..";
      await flushPromises();
      expect(wrapper.vm.namespaceNameError).toBe(
        "The name must not contain dots"
      );
    });

    it("Show validation messages", async () => {
      invalidMinAndMaxCharacters.forEach(async (character) => {
        wrapper.vm.namespaceName = character;
        await flushPromises();
        expect(wrapper.vm.namespaceNameError).toBe(
          "this must be at most 30 characters"
        );
      });
    });

    // TODO RFC1123
  });
});
