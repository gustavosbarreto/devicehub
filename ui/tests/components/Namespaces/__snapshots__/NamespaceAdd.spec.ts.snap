// Vitest Snapshot v1

exports[`NamespaceAdd > Dialog > Renders the component 1`] = `
"<!--v-if-->
<v-list-item-title-stub tag=\\"div\\"></v-list-item-title-stub>"
`;

exports[`NamespaceAdd > Doesn't render the dialog > Renders the component 1`] = `
"<!--v-if-->
<div class=\\"v-list-item-title\\">
  <!---->
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
